### PR TITLE
Refactor kotti pagination

### PR DESCRIPTION
**Pull request by Isokaeder**
The usual shtick, some typescript, some zod, some rearranging of code and whatnot

### CODE REVIEW

File: packages/kotti-ui/source/kotti-pagination/components/PaginationExpanded.vue [hunk 1]
```diff
@@ -1,47 +1,46 @@
 <template>
-	<div class="inline-container">
+	<div class="kt-pagination__inline-container">
 		<li
 			v-for="(page, index) in totalPages"
 			:key="index"
-			:class="paginatorClasses(page, 'page-item--active')"
-			@click="$emit('setPage', page)"
+			:class="paginatorClasses(page)"
+			@click="$emit('setPage', page - 1)"
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-pagination/components/PaginationExpanded.vue:7]: This looks wrong. Why was this changed?

File: packages/kotti-ui/source/kotti-pagination/KtPagination.vue [hunk 2]
```diff
@@ -1,163 +1,139 @@
...
+				currentPage: currentPage.value,
+				fixedWidth: props.fixedWidth,
+				maximumPage: maximumPage.value,
+				pageSize: props.pageSize,
+				total: props.total,
+			})),
+			previousPage: () => {
+				if (currentPage.value === 0) return
+				currentPage.value -= 1
+				emit('previousPageClicked', currentPage.value + 1)
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-pagination/KtPagination.vue:85]: This API should probably be simplified. Why would the app care if the previous button was clicked instead of a specific page number? Let’s just emit the normal event

Maybe in a follow-up PR after this one is merged, alternatively let’s make an issue if you don’t want to do this immediately

File: packages/kotti-ui/source/kotti-pagination/KtPagination.vue [hunk 3]
```diff
@@ -1,163 +1,139 @@
...
+			},
+			setCurrentPage: (page) => {
+				if (page === currentPage.value) return
+				currentPage.value = page
+				emit('currentPageChange', currentPage.value + 1)
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-pagination/KtPagination.vue:90]: This API should probably be simplified. Why would the app care if the previous button was clicked instead of a specific page number? Let’s just emit the normal event

Maybe in a follow-up PR after this one is merged, alternatively let’s make an issue if you don’t want to do this immediately

File: packages/kotti-ui/source/kotti-pagination/KtPagination.vue [hunk 4]
```diff
@@ -1,163 +1,139 @@
...
-	props: {
-		adjacentAmount: { type: Number, default: 1 },
-		fixedWidth: { type: Boolean, default: false },
-		page: { type: Number, default: 1 },
-		pageSize: { type: Number, default: 10 },
-		pagingStyle: { type: String, default: 'expand' },
-		total: { type: Number, required: true },
+	props: makeProps(KottiPagination.propsSchema),
+	setup(props, { emit }) {
+		const currentPage = ref(props.page - 1)
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-pagination/KtPagination.vue:45]: This API should probably just be controlled so we can ensure that the page is properly handled.

I’d suggest either a follow-up PR after this one is merged, or an issue if you don’t want to do this immediately

**Review by FlorianWendelborn — APPROVED** [review]

**Comment by Isokaeder** [comment]
Closed, replaced by #1007

**Comment by Isokaeder** [comment]
Closed, replaced by #1007
